### PR TITLE
feat: enable cli creation for go impl (#40)

### PR DESCRIPTION
**Pull request by lstocchi**
This is the first part of #40 so that we can build and release alizer-cli binary

I think this gives a similar experience as the quarkus cli. 
I also added a method for the devfile part to work with a registry as it's a better experience with a cli.

Once this is merged we can work on a workflow to build and release cli binaries

### CODE REVIEW

File: go/go.mod [hunk 1]
```diff
@@ -3,11 +3,50 @@ module github.com/redhat-developer/alizer/go
...
+)
+
+require (
+	github.com/inconshreveable/mousetrap v1.0.0 // indirect
 	github.com/sabhiram/go-gitignore v0.0.0-20210923224102-525f6e181f06 // indirect
-	golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898 // indirect
+	github.com/spf13/cobra v1.5.0 // indirect
+	github.com/spf13/pflag v1.0.5 // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	k8s.io/kubectl v0.24.3
```
Comment (jeffmaury) [inline go/go.mod:51]: Is this really required ? This file looks strange to me

File: go/pkg/apis/recognizer/devfile_recognizer.go [hunk 2]
```diff
@@ -62,6 +66,72 @@ func SelectDevFileUsingLanguagesFromTypes(languages []model.Language, devFileTyp
...
+	resp, err := http.Get(url)
+	if err != nil {
+		// retry by appending index to url
+		url = appendIndexPath(url)
+		resp, err = http.Get(url)
+		if err != nil {
+			return []DevFileType{}, err
+		}
+	}
+	defer resp.Body.Close()
```
Comment (jeffmaury) [inline go/pkg/apis/recognizer/devfile_recognizer.go:94]: There are HTTP responses without body are we safe here ?

**Review by jeffmaury — COMMENTED** [review]
Code looks ok but how can I build it ?

File: go/pkg/apis/recognizer/devfile_recognizer.go (hunk 2 above)
Comment (lstocchi) [reply go/pkg/apis/recognizer/devfile_recognizer.go:94]: It should be safe 
```
The http Client and Transport guarantee that Body is always non-nil, even on responses without a body or responses with a zero-length body. It is the caller's responsibility to close Body.
```

File: go/go.mod (hunk 1 above)
Comment (lstocchi) [reply go/go.mod:51]: I was trying to do something similar to odo by using ktemplates but then I changed my mind as that was not really necessary for us. I guess this was added by ktemplate. I'll clean it. Good catch :+1:

File: go/alizer.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+func main() {
+	alizerCmd := cli.AlizerCommands()
+	flag.Usage = func() {
+		_ = alizerCmd.Help()
+	}
+	// parse the flags but hack around to avoid exiting with error code 2 on help
+	flag.CommandLine.Init(os.Args[0], flag.ContinueOnError)
+	args := os.Args[1:]
+	if err := flag.CommandLine.Parse(args); err != nil {
+		if err == flag.ErrHelp {
```
[the remark this answers is not in the document]
Comment (lstocchi) [reply go/alizer.go:20]: Added a exit(1) when an unknown flag is passed. This also prevented an output issue. I could see the help output displayed twice.

**Comment by lstocchi** [comment]
> Code looks ok but how can I build it ?

`go build -o alizer` not sure if in windows it's different though. I need to verify

**Review by jeffmaury — APPROVED** [review]
LGTM but was not able to get the port number. Did I miss something ?

**Pull request merged**